tools/cpm_tpm_rpk/cpm_tpm_rpk.xml

```diff
@@ -0,0 +1,154 @@
+			<option value="\ ">Whitespace</option>
+			<option value=".">Dots</option>
+                        <option value="_">Underscores</option>
+                        <option value="-">Dashes</option>
+		</param>
+		<param name="input_header" type="select" label="Consider first line of input file as header?">
+			<option value="TRUE" selected="true">Yes</option>
+			<option value="FALSE">No</option>
+		</param>
+                <conditional name="option">
+	                <param name="type_transfo" type="select" label="Type of transformation">
+      		                <option value="cpm" selected="true">CPM</option>
+               		        <option value="tpm">TPM</option>
+               	        	<option value="rpk">RPK</option>
+	               	</param>
+			<when value="tpm">
+				<param name="gene_file" type="data" format="txt" label="Gene length file"/> 
+				<param name="gene_sep" type="select" label="Gene length column separator">
+					<option value="tab" selected="true">Tabs</option>
+   	        			<option value=",">Commas</option>
+					<option value="\ ">Whitespaces</option>
+					<option value=".">Dots</option>
+					<option value="_">Underscores</option>
+					<option value="-">Dashes</option>
+				</param>
+				<param name="gene_header" type="select" label="Consider first line of gene length file as header ?">
+					<option value="TRUE" selected="true">Yes</option>
+					<option value="FALSE">No</option>
+				</param>
+			</when>
+                        <when value="rpk">
+                                <param name="gene_file" type="data" format="txt" label="Gene length file"/>
+                                <param name="gene_sep" type="select" label="Gene length column separator">
+                                        <option value="tab" selected="true">Tabs</option>
+                                        <option value=",">Commas</option>
+                                        <option value="\ ">Whitespaces</option>
+                                        <option value=".">Dots</option>
+                                        <option value="_">Underscores</option>
+                                        <option value="-">Dashes</option>
+                                </param>
+                                <param name="gene_header" type="select" label="Consider first line of gene length file as header ?">
+                                        <option value="TRUE" selected="true">Yes</option>
+                                        <option value="FALSE">No</option>
+                                </param>
+                        </when>
+
+			<when value="cpm">
+			</when>
+		</conditional>
+		<param name="log" type="select" label="Data should be log transformed ?">
+			<option value="FALSE" selected="true">No</option>
+			<option value="TRUE">Yes</option>
+		</param>
+	</inputs>
+	<outputs>
+                <data name="output" format="tabular" label="Transformation of raw counts" />
```

Comment:
In addition to the conditional output, you can add column metadata (that may change in function of the output) using the code (from iuc DESeq2):

https://github.com/galaxyproject/tools-iuc/blob/master/tools/deseq2/deseq2.xml#L164